openedx/core/lib/xblock_builtin/xblock_discussion/xblock_discussion.py

```diff
@@ -252,3 +252,44 @@ def _apply_metadata_and_policy(cls, block, node, runtime):
         for field_name, value in metadata.iteritems():
             if field_name in block.fields:
                 setattr(block, field_name, value)
+
+
+@XBlock.needs('user')
+@XBlock.needs('i18n')
+class DiscussionBoardBlock(UIBlock):
+    """
+    Provides a UI block that can render the full discussion board.
+    """
+
+    @property
+    def course_key(self):
+        """
+        :return: int course id
+
+        NB: The goal is to move this XBlock out of edx-platform, and so we use
+        scope_ids.usage_id instead of runtime.course_id so that the code will
+        continue to work with workbench-based testing.
+        """
+        return getattr(self.scope_ids.usage_id, 'course_key', None)
```

Comment:
Yep, cut-and-paste issue. I had been thinking that we'd configure the blocks with their course, but then we settled upon passing the course key in through the context instead.